// @faustjs/core/dist/api/client/schema.generated.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generatedSchema = exports.scalarsEnumsHash = exports.MediaItemStatusEnum = exports.UsersConnectionSearchColumnEnum = exports.UserRoleEnum = exports.UsersConnectionOrderbyEnum = exports.UserNodeIdTypeEnum = exports.TermNodeIdTypeEnum = exports.TaxonomyIdTypeEnum = exports.TagIdType = exports.PostFormatIdType = exports.PostIdType = exports.PageIdType = exports.MenuItemNodeIdTypeEnum = exports.MenuLocationEnum = exports.MenuNodeIdTypeEnum = exports.MediaItemIdType = exports.ContentTypeIdTypeEnum = exports.ContentNodeIdTypeEnum = exports.CategoryIdType = exports.TaxonomyEnum = exports.ContentTypesOfTagEnum = exports.ContentTypesOfPostFormatEnum = exports.MediaItemSizeEnum = exports.PostObjectFieldFormatEnum = exports.CommentsConnectionOrderbyEnum = exports.AvatarRatingEnum = exports.ContentTypeEnum = exports.PostStatusEnum = exports.PostObjectsConnectionOrderbyEnum = exports.MimeTypeEnum = exports.RelationEnum = exports.PostObjectsConnectionDateColumnEnum = exports.ContentTypesOfCategoryEnum = exports.TermObjectsConnectionOrderbyEnum = exports.OrderEnum = void 0;
const gqty_1 = require("gqty");
var OrderEnum;
(function (OrderEnum) {
    OrderEnum["ASC"] = "ASC";
    OrderEnum["DESC"] = "DESC";
})(OrderEnum = exports.OrderEnum || (exports.OrderEnum = {}));
var TermObjectsConnectionOrderbyEnum;
(function (TermObjectsConnectionOrderbyEnum) {
    TermObjectsConnectionOrderbyEnum["COUNT"] = "COUNT";
    TermObjectsConnectionOrderbyEnum["DESCRIPTION"] = "DESCRIPTION";
    TermObjectsConnectionOrderbyEnum["NAME"] = "NAME";
    TermObjectsConnectionOrderbyEnum["SLUG"] = "SLUG";
    TermObjectsConnectionOrderbyEnum["TERM_GROUP"] = "TERM_GROUP";
    TermObjectsConnectionOrderbyEnum["TERM_ID"] = "TERM_ID";
    TermObjectsConnectionOrderbyEnum["TERM_ORDER"] = "TERM_ORDER";
})(TermObjectsConnectionOrderbyEnum = exports.TermObjectsConnectionOrderbyEnum || (exports.TermObjectsConnectionOrderbyEnum = {}));
var ContentTypesOfCategoryEnum;
(function (ContentTypesOfCategoryEnum) {
    ContentTypesOfCategoryEnum["POST"] = "POST";
})(ContentTypesOfCategoryEnum = exports.ContentTypesOfCategoryEnum || (exports.ContentTypesOfCategoryEnum = {}));
var PostObjectsConnectionDateColumnEnum;
(function (PostObjectsConnectionDateColumnEnum) {
    PostObjectsConnectionDateColumnEnum["DATE"] = "DATE";
    PostObjectsConnectionDateColumnEnum["MODIFIED"] = "MODIFIED";
})(PostObjectsConnectionDateColumnEnum = exports.PostObjectsConnectionDateColumnEnum || (exports.PostObjectsConnectionDateColumnEnum = {}));
var RelationEnum;
(function (RelationEnum) {
    RelationEnum["AND"] = "AND";
    RelationEnum["OR"] = "OR";
})(RelationEnum = exports.RelationEnum || (exports.RelationEnum = {}));
var MimeTypeEnum;
(function (MimeTypeEnum) {
    MimeTypeEnum["APPLICATION_JAVA"] = "APPLICATION_JAVA";
    MimeTypeEnum["APPLICATION_MSWORD"] = "APPLICATION_MSWORD";
    MimeTypeEnum["APPLICATION_OCTET_STREAM"] = "APPLICATION_OCTET_STREAM";
    MimeTypeEnum["APPLICATION_ONENOTE"] = "APPLICATION_ONENOTE";
    MimeTypeEnum["APPLICATION_OXPS"] = "APPLICATION_OXPS";
    MimeTypeEnum["APPLICATION_PDF"] = "APPLICATION_PDF";
    MimeTypeEnum["APPLICATION_RAR"] = "APPLICATION_RAR";
    MimeTypeEnum["APPLICATION_RTF"] = "APPLICATION_RTF";
    MimeTypeEnum["APPLICATION_TTAF_XML"] = "APPLICATION_TTAF_XML";
    MimeTypeEnum["APPLICATION_VND_APPLE_KEYNOTE"] = "APPLICATION_VND_APPLE_KEYNOTE";
    MimeTypeEnum["APPLICATION_VND_APPLE_NUMBERS"] = "APPLICATION_VND_APPLE_NUMBERS";
    MimeTypeEnum["APPLICATION_VND_APPLE_PAGES"] = "APPLICATION_VND_APPLE_PAGES";
    MimeTypeEnum["APPLICATION_VND_MS_ACCESS"] = "APPLICATION_VND_MS_ACCESS";
    MimeTypeEnum["APPLICATION_VND_MS_EXCEL"] = "APPLICATION_VND_MS_EXCEL";
    MimeTypeEnum["APPLICATION_VND_MS_EXCEL_ADDIN_MACROENABLED_12"] = "APPLICATION_VND_MS_EXCEL_ADDIN_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_EXCEL_SHEET_BINARY_MACROENABLED_12"] = "APPLICATION_VND_MS_EXCEL_SHEET_BINARY_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_EXCEL_SHEET_MACROENABLED_12"] = "APPLICATION_VND_MS_EXCEL_SHEET_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_EXCEL_TEMPLATE_MACROENABLED_12"] = "APPLICATION_VND_MS_EXCEL_TEMPLATE_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT"] = "APPLICATION_VND_MS_POWERPOINT";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT_ADDIN_MACROENABLED_12"] = "APPLICATION_VND_MS_POWERPOINT_ADDIN_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT_PRESENTATION_MACROENABLED_12"] = "APPLICATION_VND_MS_POWERPOINT_PRESENTATION_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT_SLIDESHOW_MACROENABLED_12"] = "APPLICATION_VND_MS_POWERPOINT_SLIDESHOW_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT_SLIDE_MACROENABLED_12"] = "APPLICATION_VND_MS_POWERPOINT_SLIDE_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_POWERPOINT_TEMPLATE_MACROENABLED_12"] = "APPLICATION_VND_MS_POWERPOINT_TEMPLATE_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_PROJECT"] = "APPLICATION_VND_MS_PROJECT";
    MimeTypeEnum["APPLICATION_VND_MS_WORD_DOCUMENT_MACROENABLED_12"] = "APPLICATION_VND_MS_WORD_DOCUMENT_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_WORD_TEMPLATE_MACROENABLED_12"] = "APPLICATION_VND_MS_WORD_TEMPLATE_MACROENABLED_12";
    MimeTypeEnum["APPLICATION_VND_MS_WRITE"] = "APPLICATION_VND_MS_WRITE";
    MimeTypeEnum["APPLICATION_VND_MS_XPSDOCUMENT"] = "APPLICATION_VND_MS_XPSDOCUMENT";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_CHART"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_CHART";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_DATABASE"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_DATABASE";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_FORMULA"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_FORMULA";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_GRAPHICS"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_GRAPHICS";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_PRESENTATION"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_PRESENTATION";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_SPREADSHEET"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_SPREADSHEET";
    MimeTypeEnum["APPLICATION_VND_OASIS_OPENDOCUMENT_TEXT"] = "APPLICATION_VND_OASIS_OPENDOCUMENT_TEXT";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_PRESENTATION"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_PRESENTATION";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_SLIDE"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_SLIDE";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_SLIDESHOW"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_SLIDESHOW";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_TEMPLATE"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_PRESENTATIONML_TEMPLATE";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_SPREADSHEETML_SHEET"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_SPREADSHEETML_SHEET";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_SPREADSHEETML_TEMPLATE"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_SPREADSHEETML_TEMPLATE";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_WORDPROCESSINGML_DOCUMENT"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_WORDPROCESSINGML_DOCUMENT";
    MimeTypeEnum["APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_WORDPROCESSINGML_TEMPLATE"] = "APPLICATION_VND_OPENXMLFORMATS_OFFICEDOCUMENT_WORDPROCESSINGML_TEMPLATE";
    MimeTypeEnum["APPLICATION_WORDPERFECT"] = "APPLICATION_WORDPERFECT";
    MimeTypeEnum["APPLICATION_X_7Z_COMPRESSED"] = "APPLICATION_X_7Z_COMPRESSED";
    MimeTypeEnum["APPLICATION_X_GZIP"] = "APPLICATION_X_GZIP";
    MimeTypeEnum["APPLICATION_X_TAR"] = "APPLICATION_X_TAR";
    MimeTypeEnum["APPLICATION_ZIP"] = "APPLICATION_ZIP";
    MimeTypeEnum["AUDIO_AAC"] = "AUDIO_AAC";
    MimeTypeEnum["AUDIO_FLAC"] = "AUDIO_FLAC";
    MimeTypeEnum["AUDIO_MIDI"] = "AUDIO_MIDI";
    MimeTypeEnum["AUDIO_MPEG"] = "AUDIO_MPEG";
    MimeTypeEnum["AUDIO_OGG"] = "AUDIO_OGG";
    MimeTypeEnum["AUDIO_WAV"] = "AUDIO_WAV";
    MimeTypeEnum["AUDIO_X_MATROSKA"] = "AUDIO_X_MATROSKA";
    MimeTypeEnum["AUDIO_X_MS_WAX"] = "AUDIO_X_MS_WAX";
    MimeTypeEnum["AUDIO_X_MS_WMA"] = "AUDIO_X_MS_WMA";
    MimeTypeEnum["AUDIO_X_REALAUDIO"] = "AUDIO_X_REALAUDIO";
    MimeTypeEnum["IMAGE_BMP"] = "IMAGE_BMP";
    MimeTypeEnum["IMAGE_GIF"] = "IMAGE_GIF";
    MimeTypeEnum["IMAGE_HEIC"] = "IMAGE_HEIC";
    MimeTypeEnum["IMAGE_JPEG"] = "IMAGE_JPEG";
    MimeTypeEnum["IMAGE_PNG"] = "IMAGE_PNG";
    MimeTypeEnum["IMAGE_TIFF"] = "IMAGE_TIFF";
    MimeTypeEnum["IMAGE_WEBP"] = "IMAGE_WEBP";
    MimeTypeEnum["IMAGE_X_ICON"] = "IMAGE_X_ICON";
    MimeTypeEnum["TEXT_CALENDAR"] = "TEXT_CALENDAR";
    MimeTypeEnum["TEXT_CSS"] = "TEXT_CSS";
    MimeTypeEnum["TEXT_CSV"] = "TEXT_CSV";
    MimeTypeEnum["TEXT_PLAIN"] = "TEXT_PLAIN";
    MimeTypeEnum["TEXT_RICHTEXT"] = "TEXT_RICHTEXT";
    MimeTypeEnum["TEXT_TAB_SEPARATED_VALUES"] = "TEXT_TAB_SEPARATED_VALUES";
    MimeTypeEnum["TEXT_VTT"] = "TEXT_VTT";
    MimeTypeEnum["VIDEO_3GPP"] = "VIDEO_3GPP";
    MimeTypeEnum["VIDEO_3GPP2"] = "VIDEO_3GPP2";
    MimeTypeEnum["VIDEO_AVI"] = "VIDEO_AVI";
    MimeTypeEnum["VIDEO_DIVX"] = "VIDEO_DIVX";
    MimeTypeEnum["VIDEO_MP4"] = "VIDEO_MP4";
    MimeTypeEnum["VIDEO_MPEG"] = "VIDEO_MPEG";
    MimeTypeEnum["VIDEO_OGG"] = "VIDEO_OGG";
    MimeTypeEnum["VIDEO_QUICKTIME"] = "VIDEO_QUICKTIME";
    MimeTypeEnum["VIDEO_WEBM"] = "VIDEO_WEBM";
    MimeTypeEnum["VIDEO_X_FLV"] = "VIDEO_X_FLV";
    MimeTypeEnum["VIDEO_X_MATROSKA"] = "VIDEO_X_MATROSKA";
    MimeTypeEnum["VIDEO_X_MS_ASF"] = "VIDEO_X_MS_ASF";
    MimeTypeEnum["VIDEO_X_MS_WM"] = "VIDEO_X_MS_WM";
    MimeTypeEnum["VIDEO_X_MS_WMV"] = "VIDEO_X_MS_WMV";
    MimeTypeEnum["VIDEO_X_MS_WMX"] = "VIDEO_X_MS_WMX";
})(MimeTypeEnum = exports.MimeTypeEnum || (exports.MimeTypeEnum = {}));
var PostObjectsConnectionOrderbyEnum;
(function (PostObjectsConnectionOrderbyEnum) {
    PostObjectsConnectionOrderbyEnum["AUTHOR"] = "AUTHOR";
    PostObjectsConnectionOrderbyEnum["COMMENT_COUNT"] = "COMMENT_COUNT";
    PostObjectsConnectionOrderbyEnum["DATE"] = "DATE";
    PostObjectsConnectionOrderbyEnum["IN"] = "IN";
    PostObjectsConnectionOrderbyEnum["MENU_ORDER"] = "MENU_ORDER";
    PostObjectsConnectionOrderbyEnum["MODIFIED"] = "MODIFIED";
    PostObjectsConnectionOrderbyEnum["NAME_IN"] = "NAME_IN";
    PostObjectsConnectionOrderbyEnum["PARENT"] = "PARENT";
    PostObjectsConnectionOrderbyEnum["SLUG"] = "SLUG";
    PostObjectsConnectionOrderbyEnum["TITLE"] = "TITLE";
})(PostObjectsConnectionOrderbyEnum = exports.PostObjectsConnectionOrderbyEnum || (exports.PostObjectsConnectionOrderbyEnum = {}));
var PostStatusEnum;
(function (PostStatusEnum) {
    PostStatusEnum["AUTO_DRAFT"] = "AUTO_DRAFT";
    PostStatusEnum["DRAFT"] = "DRAFT";
    PostStatusEnum["FUTURE"] = "FUTURE";
    PostStatusEnum["INHERIT"] = "INHERIT";
    PostStatusEnum["PENDING"] = "PENDING";
    PostStatusEnum["PRIVATE"] = "PRIVATE";
    PostStatusEnum["PUBLISH"] = "PUBLISH";
    PostStatusEnum["REQUEST_COMPLETED"] = "REQUEST_COMPLETED";
    PostStatusEnum["REQUEST_CONFIRMED"] = "REQUEST_CONFIRMED";
    PostStatusEnum["REQUEST_FAILED"] = "REQUEST_FAILED";
    PostStatusEnum["REQUEST_PENDING"] = "REQUEST_PENDING";
    PostStatusEnum["TRASH"] = "TRASH";
})(PostStatusEnum = exports.PostStatusEnum || (exports.PostStatusEnum = {}));
var ContentTypeEnum;
(function (ContentTypeEnum) {
    ContentTypeEnum["ATTACHMENT"] = "ATTACHMENT";
    ContentTypeEnum["PAGE"] = "PAGE";
    ContentTypeEnum["POST"] = "POST";
})(ContentTypeEnum = exports.ContentTypeEnum || (exports.ContentTypeEnum = {}));
var AvatarRatingEnum;
(function (AvatarRatingEnum) {
    AvatarRatingEnum["G"] = "G";
    AvatarRatingEnum["PG"] = "PG";
    AvatarRatingEnum["R"] = "R";
    AvatarRatingEnum["X"] = "X";
})(AvatarRatingEnum = exports.AvatarRatingEnum || (exports.AvatarRatingEnum = {}));
var CommentsConnectionOrderbyEnum;
(function (CommentsConnectionOrderbyEnum) {
    CommentsConnectionOrderbyEnum["COMMENT_AGENT"] = "COMMENT_AGENT";
    CommentsConnectionOrderbyEnum["COMMENT_APPROVED"] = "COMMENT_APPROVED";
    CommentsConnectionOrderbyEnum["COMMENT_AUTHOR"] = "COMMENT_AUTHOR";
    CommentsConnectionOrderbyEnum["COMMENT_AUTHOR_EMAIL"] = "COMMENT_AUTHOR_EMAIL";
    CommentsConnectionOrderbyEnum["COMMENT_AUTHOR_IP"] = "COMMENT_AUTHOR_IP";
    CommentsConnectionOrderbyEnum["COMMENT_AUTHOR_URL"] = "COMMENT_AUTHOR_URL";
    CommentsConnectionOrderbyEnum["COMMENT_CONTENT"] = "COMMENT_CONTENT";
    CommentsConnectionOrderbyEnum["COMMENT_DATE"] = "COMMENT_DATE";
    CommentsConnectionOrderbyEnum["COMMENT_DATE_GMT"] = "COMMENT_DATE_GMT";
    CommentsConnectionOrderbyEnum["COMMENT_ID"] = "COMMENT_ID";
    CommentsConnectionOrderbyEnum["COMMENT_IN"] = "COMMENT_IN";
    CommentsConnectionOrderbyEnum["COMMENT_KARMA"] = "COMMENT_KARMA";
    CommentsConnectionOrderbyEnum["COMMENT_PARENT"] = "COMMENT_PARENT";
    CommentsConnectionOrderbyEnum["COMMENT_POST_ID"] = "COMMENT_POST_ID";
    CommentsConnectionOrderbyEnum["COMMENT_TYPE"] = "COMMENT_TYPE";
    CommentsConnectionOrderbyEnum["USER_ID"] = "USER_ID";
})(CommentsConnectionOrderbyEnum = exports.CommentsConnectionOrderbyEnum || (exports.CommentsConnectionOrderbyEnum = {}));
var PostObjectFieldFormatEnum;
(function (PostObjectFieldFormatEnum) {
    PostObjectFieldFormatEnum["RAW"] = "RAW";
    PostObjectFieldFormatEnum["RENDERED"] = "RENDERED";
})(PostObjectFieldFormatEnum = exports.PostObjectFieldFormatEnum || (exports.PostObjectFieldFormatEnum = {}));
var MediaItemSizeEnum;
(function (MediaItemSizeEnum) {
    MediaItemSizeEnum["LARGE"] = "LARGE";
    MediaItemSizeEnum["MEDIUM"] = "MEDIUM";
    MediaItemSizeEnum["MEDIUM_LARGE"] = "MEDIUM_LARGE";
    MediaItemSizeEnum["POST_THUMBNAIL"] = "POST_THUMBNAIL";
    MediaItemSizeEnum["THUMBNAIL"] = "THUMBNAIL";
    MediaItemSizeEnum["_1536X1536"] = "_1536X1536";
    MediaItemSizeEnum["_2048X2048"] = "_2048X2048";
})(MediaItemSizeEnum = exports.MediaItemSizeEnum || (exports.MediaItemSizeEnum = {}));
var ContentTypesOfPostFormatEnum;
(function (ContentTypesOfPostFormatEnum) {
    ContentTypesOfPostFormatEnum["POST"] = "POST";
})(ContentTypesOfPostFormatEnum = exports.ContentTypesOfPostFormatEnum || (exports.ContentTypesOfPostFormatEnum = {}));
var ContentTypesOfTagEnum;
(function (ContentTypesOfTagEnum) {
    ContentTypesOfTagEnum["POST"] = "POST";
})(ContentTypesOfTagEnum = exports.ContentTypesOfTagEnum || (exports.ContentTypesOfTagEnum = {}));
var TaxonomyEnum;
(function (TaxonomyEnum) {
    TaxonomyEnum["CATEGORY"] = "CATEGORY";
    TaxonomyEnum["POSTFORMAT"] = "POSTFORMAT";
    TaxonomyEnum["TAG"] = "TAG";
})(TaxonomyEnum = exports.TaxonomyEnum || (exports.TaxonomyEnum = {}));
var CategoryIdType;
(function (CategoryIdType) {
    CategoryIdType["DATABASE_ID"] = "DATABASE_ID";
    CategoryIdType["ID"] = "ID";
    CategoryIdType["NAME"] = "NAME";
    CategoryIdType["SLUG"] = "SLUG";
    CategoryIdType["URI"] = "URI";
})(CategoryIdType = exports.CategoryIdType || (exports.CategoryIdType = {}));
var ContentNodeIdTypeEnum;
(function (ContentNodeIdTypeEnum) {
    ContentNodeIdTypeEnum["DATABASE_ID"] = "DATABASE_ID";
    ContentNodeIdTypeEnum["ID"] = "ID";
    ContentNodeIdTypeEnum["URI"] = "URI";
})(ContentNodeIdTypeEnum = exports.ContentNodeIdTypeEnum || (exports.ContentNodeIdTypeEnum = {}));
var ContentTypeIdTypeEnum;
(function (ContentTypeIdTypeEnum) {
    ContentTypeIdTypeEnum["ID"] = "ID";
    ContentTypeIdTypeEnum["NAME"] = "NAME";
})(ContentTypeIdTypeEnum = exports.ContentTypeIdTypeEnum || (exports.ContentTypeIdTypeEnum = {}));
var MediaItemIdType;
(function (MediaItemIdType) {
    MediaItemIdType["DATABASE_ID"] = "DATABASE_ID";
    MediaItemIdType["ID"] = "ID";
    MediaItemIdType["SLUG"] = "SLUG";
    MediaItemIdType["SOURCE_URL"] = "SOURCE_URL";
    MediaItemIdType["URI"] = "URI";
})(MediaItemIdType = exports.MediaItemIdType || (exports.MediaItemIdType = {}));
var MenuNodeIdTypeEnum;
(function (MenuNodeIdTypeEnum) {
    MenuNodeIdTypeEnum["DATABASE_ID"] = "DATABASE_ID";
    MenuNodeIdTypeEnum["ID"] = "ID";
    MenuNodeIdTypeEnum["NAME"] = "NAME";
})(MenuNodeIdTypeEnum = exports.MenuNodeIdTypeEnum || (exports.MenuNodeIdTypeEnum = {}));
var MenuLocationEnum;
(function (MenuLocationEnum) {
    MenuLocationEnum["FOOTER"] = "FOOTER";
    MenuLocationEnum["PRIMARY"] = "PRIMARY";
})(MenuLocationEnum = exports.MenuLocationEnum || (exports.MenuLocationEnum = {}));
var MenuItemNodeIdTypeEnum;
(function (MenuItemNodeIdTypeEnum) {
    MenuItemNodeIdTypeEnum["DATABASE_ID"] = "DATABASE_ID";
    MenuItemNodeIdTypeEnum["ID"] = "ID";
})(MenuItemNodeIdTypeEnum = exports.MenuItemNodeIdTypeEnum || (exports.MenuItemNodeIdTypeEnum = {}));
var PageIdType;
(function (PageIdType) {
    PageIdType["DATABASE_ID"] = "DATABASE_ID";
    PageIdType["ID"] = "ID";
    PageIdType["URI"] = "URI";
})(PageIdType = exports.PageIdType || (exports.PageIdType = {}));
var PostIdType;
(function (PostIdType) {
    PostIdType["DATABASE_ID"] = "DATABASE_ID";
    PostIdType["ID"] = "ID";
    PostIdType["SLUG"] = "SLUG";
    PostIdType["URI"] = "URI";
})(PostIdType = exports.PostIdType || (exports.PostIdType = {}));
var PostFormatIdType;
(function (PostFormatIdType) {
    PostFormatIdType["DATABASE_ID"] = "DATABASE_ID";
    PostFormatIdType["ID"] = "ID";
    PostFormatIdType["NAME"] = "NAME";
    PostFormatIdType["SLUG"] = "SLUG";
    PostFormatIdType["URI"] = "URI";
})(PostFormatIdType = exports.PostFormatIdType || (exports.PostFormatIdType = {}));
var TagIdType;
(function (TagIdType) {
    TagIdType["DATABASE_ID"] = "DATABASE_ID";
    TagIdType["ID"] = "ID";
    TagIdType["NAME"] = "NAME";
    TagIdType["SLUG"] = "SLUG";
    TagIdType["URI"] = "URI";
})(TagIdType = exports.TagIdType || (exports.TagIdType = {}));
var TaxonomyIdTypeEnum;
(function (TaxonomyIdTypeEnum) {
    TaxonomyIdTypeEnum["ID"] = "ID";
    TaxonomyIdTypeEnum["NAME"] = "NAME";
})(TaxonomyIdTypeEnum = exports.TaxonomyIdTypeEnum || (exports.TaxonomyIdTypeEnum = {}));
var TermNodeIdTypeEnum;
(function (TermNodeIdTypeEnum) {
    TermNodeIdTypeEnum["DATABASE_ID"] = "DATABASE_ID";
    TermNodeIdTypeEnum["ID"] = "ID";
    TermNodeIdTypeEnum["NAME"] = "NAME";
    TermNodeIdTypeEnum["SLUG"] = "SLUG";
    TermNodeIdTypeEnum["URI"] = "URI";
})(TermNodeIdTypeEnum = exports.TermNodeIdTypeEnum || (exports.TermNodeIdTypeEnum = {}));
var UserNodeIdTypeEnum;
(function (UserNodeIdTypeEnum) {
    UserNodeIdTypeEnum["DATABASE_ID"] = "DATABASE_ID";
    UserNodeIdTypeEnum["EMAIL"] = "EMAIL";
    UserNodeIdTypeEnum["ID"] = "ID";
    UserNodeIdTypeEnum["SLUG"] = "SLUG";
    UserNodeIdTypeEnum["URI"] = "URI";
    UserNodeIdTypeEnum["USERNAME"] = "USERNAME";
})(UserNodeIdTypeEnum = exports.UserNodeIdTypeEnum || (exports.UserNodeIdTypeEnum = {}));
var UsersConnectionOrderbyEnum;
(function (UsersConnectionOrderbyEnum) {
    UsersConnectionOrderbyEnum["DISPLAY_NAME"] = "DISPLAY_NAME";
    UsersConnectionOrderbyEnum["EMAIL"] = "EMAIL";
    UsersConnectionOrderbyEnum["LOGIN"] = "LOGIN";
    UsersConnectionOrderbyEnum["LOGIN_IN"] = "LOGIN_IN";
    UsersConnectionOrderbyEnum["NICE_NAME"] = "NICE_NAME";
    UsersConnectionOrderbyEnum["NICE_NAME_IN"] = "NICE_NAME_IN";
    UsersConnectionOrderbyEnum["REGISTERED"] = "REGISTERED";
    UsersConnectionOrderbyEnum["URL"] = "URL";
})(UsersConnectionOrderbyEnum = exports.UsersConnectionOrderbyEnum || (exports.UsersConnectionOrderbyEnum = {}));
var UserRoleEnum;
(function (UserRoleEnum) {
    UserRoleEnum["ADMINISTRATOR"] = "ADMINISTRATOR";
    UserRoleEnum["AUTHOR"] = "AUTHOR";
    UserRoleEnum["CONTRIBUTOR"] = "CONTRIBUTOR";
    UserRoleEnum["EDITOR"] = "EDITOR";
    UserRoleEnum["SUBSCRIBER"] = "SUBSCRIBER";
})(UserRoleEnum = exports.UserRoleEnum || (exports.UserRoleEnum = {}));
var UsersConnectionSearchColumnEnum;
(function (UsersConnectionSearchColumnEnum) {
    UsersConnectionSearchColumnEnum["EMAIL"] = "EMAIL";
    UsersConnectionSearchColumnEnum["ID"] = "ID";
    UsersConnectionSearchColumnEnum["LOGIN"] = "LOGIN";
    UsersConnectionSearchColumnEnum["NICENAME"] = "NICENAME";
    UsersConnectionSearchColumnEnum["URL"] = "URL";
})(UsersConnectionSearchColumnEnum = exports.UsersConnectionSearchColumnEnum || (exports.UsersConnectionSearchColumnEnum = {}));
var MediaItemStatusEnum;
(function (MediaItemStatusEnum) {
    MediaItemStatusEnum["AUTO_DRAFT"] = "AUTO_DRAFT";
    MediaItemStatusEnum["INHERIT"] = "INHERIT";
    MediaItemStatusEnum["PRIVATE"] = "PRIVATE";
    MediaItemStatusEnum["TRASH"] = "TRASH";
})(MediaItemStatusEnum = exports.MediaItemStatusEnum || (exports.MediaItemStatusEnum = {}));
exports.scalarsEnumsHash = {
    String: true,
    Int: true,
    Boolean: true,
    ID: true,
    OrderEnum: true,
    TermObjectsConnectionOrderbyEnum: true,
    ContentTypesOfCategoryEnum: true,
    PostObjectsConnectionDateColumnEnum: true,
    RelationEnum: true,
    MimeTypeEnum: true,
    PostObjectsConnectionOrderbyEnum: true,
    PostStatusEnum: true,
    ContentTypeEnum: true,
    AvatarRatingEnum: true,
    CommentsConnectionOrderbyEnum: true,
    PostObjectFieldFormatEnum: true,
    MediaItemSizeEnum: true,
    Float: true,
    ContentTypesOfPostFormatEnum: true,
    ContentTypesOfTagEnum: true,
    TaxonomyEnum: true,
    CategoryIdType: true,
    ContentNodeIdTypeEnum: true,
    ContentTypeIdTypeEnum: true,
    MediaItemIdType: true,
    MenuNodeIdTypeEnum: true,
    MenuLocationEnum: true,
    MenuItemNodeIdTypeEnum: true,
    PageIdType: true,
    PostIdType: true,
    PostFormatIdType: true,
    TagIdType: true,
    TaxonomyIdTypeEnum: true,
    TermNodeIdTypeEnum: true,
    UserNodeIdTypeEnum: true,
    UsersConnectionOrderbyEnum: true,
    UserRoleEnum: true,
    UsersConnectionSearchColumnEnum: true,
    MediaItemStatusEnum: true,
};
exports.generatedSchema = {
    query: {
        __typename: { __type: 'String!' },
        allSettings: { __type: 'Settings' },
        categories: {
            __type: 'RootQueryToCategoryConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToCategoryConnectionWhereArgs',
            },
        },
        category: {
            __type: 'Category',
            __args: { id: 'ID!', idType: 'CategoryIdType' },
        },
        comment: { __type: 'Comment', __args: { id: 'ID!' } },
        comments: {
            __type: 'RootQueryToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToCommentConnectionWhereArgs',
            },
        },
        contentNode: {
            __type: 'ContentNode',
            __args: {
                id: 'ID!',
                idType: 'ContentNodeIdTypeEnum',
                contentType: 'ContentTypeEnum',
                asPreview: 'Boolean',
            },
        },
        contentNodes: {
            __type: 'RootQueryToContentNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToContentNodeConnectionWhereArgs',
            },
        },
        contentType: {
            __type: 'ContentType',
            __args: { id: 'ID!', idType: 'ContentTypeIdTypeEnum' },
        },
        contentTypes: {
            __type: 'RootQueryToContentTypeConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        discussionSettings: { __type: 'DiscussionSettings' },
        generalSettings: { __type: 'GeneralSettings' },
        mediaItem: {
            __type: 'MediaItem',
            __args: { id: 'ID!', idType: 'MediaItemIdType', asPreview: 'Boolean' },
        },
        mediaItemBy: {
            __type: 'MediaItem',
            __args: { id: 'ID', mediaItemId: 'Int', uri: 'String', slug: 'String' },
        },
        mediaItems: {
            __type: 'RootQueryToMediaItemConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToMediaItemConnectionWhereArgs',
            },
        },
        menu: {
            __type: 'Menu',
            __args: { id: 'ID!', idType: 'MenuNodeIdTypeEnum' },
        },
        menuItem: {
            __type: 'MenuItem',
            __args: { id: 'ID!', idType: 'MenuItemNodeIdTypeEnum' },
        },
        menuItems: {
            __type: 'RootQueryToMenuItemConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToMenuItemConnectionWhereArgs',
            },
        },
        menus: {
            __type: 'RootQueryToMenuConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToMenuConnectionWhereArgs',
            },
        },
        node: { __type: 'Node', __args: { id: 'ID' } },
        nodeByUri: {
            __type: 'UniformResourceIdentifiable',
            __args: { uri: 'String!' },
        },
        page: {
            __type: 'Page',
            __args: { id: 'ID!', idType: 'PageIdType', asPreview: 'Boolean' },
        },
        pageBy: {
            __type: 'Page',
            __args: { id: 'ID', pageId: 'Int', uri: 'String' },
        },
        pages: {
            __type: 'RootQueryToPageConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToPageConnectionWhereArgs',
            },
        },
        plugin: { __type: 'Plugin', __args: { id: 'ID!' } },
        plugins: {
            __type: 'RootQueryToPluginConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        post: {
            __type: 'Post',
            __args: { id: 'ID!', idType: 'PostIdType', asPreview: 'Boolean' },
        },
        postBy: {
            __type: 'Post',
            __args: { id: 'ID', postId: 'Int', uri: 'String', slug: 'String' },
        },
        postFormat: {
            __type: 'PostFormat',
            __args: { id: 'ID!', idType: 'PostFormatIdType' },
        },
        postFormats: {
            __type: 'RootQueryToPostFormatConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToPostFormatConnectionWhereArgs',
            },
        },
        posts: {
            __type: 'RootQueryToPostConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToPostConnectionWhereArgs',
            },
        },
        readingSettings: { __type: 'ReadingSettings' },
        registeredScripts: {
            __type: 'RootQueryToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        registeredStylesheets: {
            __type: 'RootQueryToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        revisions: {
            __type: 'RootQueryToContentRevisionUnionConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToContentRevisionUnionConnectionWhereArgs',
            },
        },
        tag: { __type: 'Tag', __args: { id: 'ID!', idType: 'TagIdType' } },
        tags: {
            __type: 'RootQueryToTagConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToTagConnectionWhereArgs',
            },
        },
        taxonomies: {
            __type: 'RootQueryToTaxonomyConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        taxonomy: {
            __type: 'Taxonomy',
            __args: { id: 'ID!', idType: 'TaxonomyIdTypeEnum' },
        },
        termNode: {
            __type: 'TermNode',
            __args: {
                id: 'ID!',
                idType: 'TermNodeIdTypeEnum',
                taxonomy: 'TaxonomyEnum',
            },
        },
        terms: {
            __type: 'RootQueryToTermNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToTermNodeConnectionWhereArgs',
            },
        },
        theme: { __type: 'Theme', __args: { id: 'ID!' } },
        themes: {
            __type: 'RootQueryToThemeConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        user: {
            __type: 'User',
            __args: { id: 'ID!', idType: 'UserNodeIdTypeEnum' },
        },
        userRole: { __type: 'UserRole', __args: { id: 'ID!' } },
        userRoles: {
            __type: 'RootQueryToUserRoleConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        users: {
            __type: 'RootQueryToUserConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'RootQueryToUserConnectionWhereArgs',
            },
        },
        viewer: { __type: 'User' },
        writingSettings: { __type: 'WritingSettings' },
    },
    mutation: {
        __typename: { __type: 'String!' },
        createCategory: {
            __type: 'CreateCategoryPayload',
            __args: { input: 'CreateCategoryInput!' },
        },
        createComment: {
            __type: 'CreateCommentPayload',
            __args: { input: 'CreateCommentInput!' },
        },
        createMediaItem: {
            __type: 'CreateMediaItemPayload',
            __args: { input: 'CreateMediaItemInput!' },
        },
        createPage: {
            __type: 'CreatePagePayload',
            __args: { input: 'CreatePageInput!' },
        },
        createPost: {
            __type: 'CreatePostPayload',
            __args: { input: 'CreatePostInput!' },
        },
        createPostFormat: {
            __type: 'CreatePostFormatPayload',
            __args: { input: 'CreatePostFormatInput!' },
        },
        createTag: {
            __type: 'CreateTagPayload',
            __args: { input: 'CreateTagInput!' },
        },
        createUser: {
            __type: 'CreateUserPayload',
            __args: { input: 'CreateUserInput!' },
        },
        deleteCategory: {
            __type: 'DeleteCategoryPayload',
            __args: { input: 'DeleteCategoryInput!' },
        },
        deleteComment: {
            __type: 'DeleteCommentPayload',
            __args: { input: 'DeleteCommentInput!' },
        },
        deleteMediaItem: {
            __type: 'DeleteMediaItemPayload',
            __args: { input: 'DeleteMediaItemInput!' },
        },
        deletePage: {
            __type: 'DeletePagePayload',
            __args: { input: 'DeletePageInput!' },
        },
        deletePost: {
            __type: 'DeletePostPayload',
            __args: { input: 'DeletePostInput!' },
        },
        deletePostFormat: {
            __type: 'DeletePostFormatPayload',
            __args: { input: 'DeletePostFormatInput!' },
        },
        deleteTag: {
            __type: 'DeleteTagPayload',
            __args: { input: 'DeleteTagInput!' },
        },
        deleteUser: {
            __type: 'DeleteUserPayload',
            __args: { input: 'DeleteUserInput!' },
        },
        increaseCount: { __type: 'Int', __args: { count: 'Int' } },
        registerUser: {
            __type: 'RegisterUserPayload',
            __args: { input: 'RegisterUserInput!' },
        },
        resetUserPassword: {
            __type: 'ResetUserPasswordPayload',
            __args: { input: 'ResetUserPasswordInput!' },
        },
        restoreComment: {
            __type: 'RestoreCommentPayload',
            __args: { input: 'RestoreCommentInput!' },
        },
        sendPasswordResetEmail: {
            __type: 'SendPasswordResetEmailPayload',
            __args: { input: 'SendPasswordResetEmailInput!' },
        },
        updateCategory: {
            __type: 'UpdateCategoryPayload',
            __args: { input: 'UpdateCategoryInput!' },
        },
        updateComment: {
            __type: 'UpdateCommentPayload',
            __args: { input: 'UpdateCommentInput!' },
        },
        updateMediaItem: {
            __type: 'UpdateMediaItemPayload',
            __args: { input: 'UpdateMediaItemInput!' },
        },
        updatePage: {
            __type: 'UpdatePagePayload',
            __args: { input: 'UpdatePageInput!' },
        },
        updatePost: {
            __type: 'UpdatePostPayload',
            __args: { input: 'UpdatePostInput!' },
        },
        updatePostFormat: {
            __type: 'UpdatePostFormatPayload',
            __args: { input: 'UpdatePostFormatInput!' },
        },
        updateSettings: {
            __type: 'UpdateSettingsPayload',
            __args: { input: 'UpdateSettingsInput!' },
        },
        updateTag: {
            __type: 'UpdateTagPayload',
            __args: { input: 'UpdateTagInput!' },
        },
        updateUser: {
            __type: 'UpdateUserPayload',
            __args: { input: 'UpdateUserInput!' },
        },
    },
    subscription: {},
    Settings: {
        __typename: { __type: 'String!' },
        discussionSettingsDefaultCommentStatus: { __type: 'String' },
        discussionSettingsDefaultPingStatus: { __type: 'String' },
        generalSettingsDateFormat: { __type: 'String' },
        generalSettingsDescription: { __type: 'String' },
        generalSettingsEmail: { __type: 'String' },
        generalSettingsLanguage: { __type: 'String' },
        generalSettingsStartOfWeek: { __type: 'Int' },
        generalSettingsTimeFormat: { __type: 'String' },
        generalSettingsTimezone: { __type: 'String' },
        generalSettingsTitle: { __type: 'String' },
        generalSettingsUrl: { __type: 'String' },
        readingSettingsPostsPerPage: { __type: 'Int' },
        writingSettingsDefaultCategory: { __type: 'Int' },
        writingSettingsDefaultPostFormat: { __type: 'String' },
        writingSettingsUseSmilies: { __type: 'Boolean' },
    },
    RootQueryToCategoryConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    RootQueryToCategoryConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToCategoryConnectionEdge]' },
        nodes: { __type: '[Category]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToCategoryConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Category' },
    },
    Category: {
        __typename: { __type: 'String!' },
        ancestors: {
            __type: 'CategoryToAncestorsCategoryConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        categoryId: { __type: 'Int' },
        children: {
            __type: 'CategoryToCategoryConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'CategoryToCategoryConnectionWhereArgs',
            },
        },
        contentNodes: {
            __type: 'CategoryToContentNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'CategoryToContentNodeConnectionWhereArgs',
            },
        },
        count: { __type: 'Int' },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        enqueuedScripts: {
            __type: 'TermNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'TermNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        link: { __type: 'String' },
        name: { __type: 'String' },
        parent: { __type: 'CategoryToParentCategoryConnectionEdge' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
        posts: {
            __type: 'CategoryToPostConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'CategoryToPostConnectionWhereArgs',
            },
        },
        slug: { __type: 'String' },
        taxonomy: { __type: 'CategoryToTaxonomyConnectionEdge' },
        termGroupId: { __type: 'Int' },
        termTaxonomyId: { __type: 'Int' },
        uri: { __type: 'String' },
    },
    Node: { __typename: { __type: 'String!' }, id: { __type: 'ID!' } },
    TermNode: {
        __typename: { __type: 'String!' },
        count: { __type: 'Int' },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        enqueuedScripts: {
            __type: 'TermNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'TermNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        link: { __type: 'String' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
        termGroupId: { __type: 'Int' },
        termTaxonomyId: { __type: 'Int' },
        uri: { __type: 'String' },
    },
    UniformResourceIdentifiable: {
        __typename: { __type: 'String!' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isTermNode: { __type: 'Boolean!' },
        uri: { __type: 'String' },
    },
    TermNodeToEnqueuedScriptConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[TermNodeToEnqueuedScriptConnectionEdge]' },
        nodes: { __type: '[EnqueuedScript]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    TermNodeToEnqueuedScriptConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedScript' },
    },
    EnqueuedScript: {
        __typename: { __type: 'String!' },
        args: { __type: 'Boolean' },
        dependencies: { __type: '[EnqueuedScript]' },
        extra: { __type: 'String' },
        handle: { __type: 'String' },
        id: { __type: 'ID!' },
        src: { __type: 'String' },
        version: { __type: 'String' },
    },
    EnqueuedAsset: {
        __typename: { __type: 'String!' },
        args: { __type: 'Boolean' },
        dependencies: { __type: '[EnqueuedScript]' },
        extra: { __type: 'String' },
        handle: { __type: 'String' },
        id: { __type: 'ID!' },
        src: { __type: 'String' },
        version: { __type: 'String' },
    },
    WPPageInfo: {
        __typename: { __type: 'String!' },
        endCursor: { __type: 'String' },
        hasNextPage: { __type: 'Boolean!' },
        hasPreviousPage: { __type: 'Boolean!' },
        startCursor: { __type: 'String' },
    },
    TermNodeToEnqueuedStylesheetConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[TermNodeToEnqueuedStylesheetConnectionEdge]' },
        nodes: { __type: '[EnqueuedStylesheet]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    TermNodeToEnqueuedStylesheetConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedStylesheet' },
    },
    EnqueuedStylesheet: {
        __typename: { __type: 'String!' },
        args: { __type: 'Boolean' },
        dependencies: { __type: '[EnqueuedScript]' },
        extra: { __type: 'String' },
        handle: { __type: 'String' },
        id: { __type: 'ID!' },
        src: { __type: 'String' },
        version: { __type: 'String' },
    },
    DatabaseIdentifier: {
        __typename: { __type: 'String!' },
        databaseId: { __type: 'Int!' },
    },
    HierarchicalTermNode: {
        __typename: { __type: 'String!' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
    },
    MenuItemLinkable: {
        __typename: { __type: 'String!' },
        databaseId: { __type: 'Int!' },
        id: { __type: 'ID!' },
        uri: { __type: 'String' },
    },
    CategoryToAncestorsCategoryConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[CategoryToAncestorsCategoryConnectionEdge]' },
        nodes: { __type: '[Category]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    CategoryToAncestorsCategoryConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Category' },
    },
    CategoryToCategoryConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    CategoryToCategoryConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[CategoryToCategoryConnectionEdge]' },
        nodes: { __type: '[Category]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    CategoryToCategoryConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Category' },
    },
    CategoryToContentNodeConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypesOfCategoryEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    DateQueryInput: {
        after: { __type: 'DateInput' },
        before: { __type: 'DateInput' },
        column: { __type: 'PostObjectsConnectionDateColumnEnum' },
        compare: { __type: 'String' },
        day: { __type: 'Int' },
        hour: { __type: 'Int' },
        inclusive: { __type: 'Boolean' },
        minute: { __type: 'Int' },
        month: { __type: 'Int' },
        relation: { __type: 'RelationEnum' },
        second: { __type: 'Int' },
        week: { __type: 'Int' },
        year: { __type: 'Int' },
    },
    DateInput: {
        day: { __type: 'Int' },
        month: { __type: 'Int' },
        year: { __type: 'Int' },
    },
    PostObjectsConnectionOrderbyInput: {
        field: { __type: 'PostObjectsConnectionOrderbyEnum!' },
        order: { __type: 'OrderEnum!' },
    },
    CategoryToContentNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[CategoryToContentNodeConnectionEdge]' },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    CategoryToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    ContentNode: {
        __typename: { __type: 'String!' },
        contentType: { __type: 'ContentNodeToContentTypeConnectionEdge' },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        desiredSlug: { __type: 'String' },
        editingLockedBy: { __type: 'ContentNodeToEditLockConnectionEdge' },
        enclosure: { __type: 'String' },
        enqueuedScripts: {
            __type: 'ContentNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'ContentNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        guid: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isPreview: { __type: 'Boolean' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        lastEditedBy: { __type: 'ContentNodeToEditLastConnectionEdge' },
        link: { __type: 'String' },
        modified: { __type: 'String' },
        modifiedGmt: { __type: 'String' },
        previewRevisionDatabaseId: { __type: 'Int' },
        previewRevisionId: { __type: 'ID' },
        slug: { __type: 'String' },
        status: { __type: 'String' },
        template: { __type: 'ContentTemplate' },
        uri: { __type: 'String' },
    },
    ContentNodeToContentTypeConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'ContentType' },
    },
    ContentType: {
        __typename: { __type: 'String!' },
        canExport: { __type: 'Boolean' },
        connectedTaxonomies: {
            __type: 'ContentTypeToTaxonomyConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        contentNodes: {
            __type: 'ContentTypeToContentNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'ContentTypeToContentNodeConnectionWhereArgs',
            },
        },
        deleteWithUser: { __type: 'Boolean' },
        description: { __type: 'String' },
        excludeFromSearch: { __type: 'Boolean' },
        graphqlPluralName: { __type: 'String' },
        graphqlSingleName: { __type: 'String' },
        hasArchive: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isFrontPage: { __type: 'Boolean!' },
        isPostsPage: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        label: { __type: 'String' },
        labels: { __type: 'PostTypeLabelDetails' },
        menuIcon: { __type: 'String' },
        menuPosition: { __type: 'Int' },
        name: { __type: 'String' },
        public: { __type: 'Boolean' },
        publiclyQueryable: { __type: 'Boolean' },
        restBase: { __type: 'String' },
        restControllerClass: { __type: 'String' },
        showInAdminBar: { __type: 'Boolean' },
        showInGraphql: { __type: 'Boolean' },
        showInMenu: { __type: 'Boolean' },
        showInNavMenus: { __type: 'Boolean' },
        showInRest: { __type: 'Boolean' },
        showUi: { __type: 'Boolean' },
        uri: { __type: 'String' },
    },
    ContentTypeToTaxonomyConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[ContentTypeToTaxonomyConnectionEdge]' },
        nodes: { __type: '[Taxonomy]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    ContentTypeToTaxonomyConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Taxonomy' },
    },
    Taxonomy: {
        __typename: { __type: 'String!' },
        connectedContentTypes: {
            __type: 'TaxonomyToContentTypeConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        description: { __type: 'String' },
        graphqlPluralName: { __type: 'String' },
        graphqlSingleName: { __type: 'String' },
        hierarchical: { __type: 'Boolean' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        label: { __type: 'String' },
        name: { __type: 'String' },
        public: { __type: 'Boolean' },
        restBase: { __type: 'String' },
        restControllerClass: { __type: 'String' },
        showCloud: { __type: 'Boolean' },
        showInAdminColumn: { __type: 'Boolean' },
        showInGraphql: { __type: 'Boolean' },
        showInMenu: { __type: 'Boolean' },
        showInNavMenus: { __type: 'Boolean' },
        showInQuickEdit: { __type: 'Boolean' },
        showInRest: { __type: 'Boolean' },
        showUi: { __type: 'Boolean' },
    },
    TaxonomyToContentTypeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[TaxonomyToContentTypeConnectionEdge]' },
        nodes: { __type: '[ContentType]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    TaxonomyToContentTypeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentType' },
    },
    ContentTypeToContentNodeConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    ContentTypeToContentNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[ContentTypeToContentNodeConnectionEdge]' },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    ContentTypeToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    PostTypeLabelDetails: {
        __typename: { __type: 'String!' },
        addNew: { __type: 'String' },
        addNewItem: { __type: 'String' },
        allItems: { __type: 'String' },
        archives: { __type: 'String' },
        attributes: { __type: 'String' },
        editItem: { __type: 'String' },
        featuredImage: { __type: 'String' },
        filterItemsList: { __type: 'String' },
        insertIntoItem: { __type: 'String' },
        itemsList: { __type: 'String' },
        itemsListNavigation: { __type: 'String' },
        menuName: { __type: 'String' },
        name: { __type: 'String' },
        newItem: { __type: 'String' },
        notFound: { __type: 'String' },
        notFoundInTrash: { __type: 'String' },
        parentItemColon: { __type: 'String' },
        removeFeaturedImage: { __type: 'String' },
        searchItems: { __type: 'String' },
        setFeaturedImage: { __type: 'String' },
        singularName: { __type: 'String' },
        uploadedToThisItem: { __type: 'String' },
        useFeaturedImage: { __type: 'String' },
        viewItem: { __type: 'String' },
        viewItems: { __type: 'String' },
    },
    ContentNodeToEditLockConnectionEdge: {
        __typename: { __type: 'String!' },
        lockTimestamp: { __type: 'String' },
        node: { __type: 'User' },
    },
    User: {
        __typename: { __type: 'String!' },
        avatar: {
            __type: 'Avatar',
            __args: {
                size: 'Int',
                forceDefault: 'Boolean',
                rating: 'AvatarRatingEnum',
            },
        },
        capKey: { __type: 'String' },
        capabilities: { __type: '[String]' },
        comments: {
            __type: 'UserToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'UserToCommentConnectionWhereArgs',
            },
        },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        email: { __type: 'String' },
        enqueuedScripts: {
            __type: 'UserToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'UserToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        extraCapabilities: { __type: '[String]' },
        firstName: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        lastName: { __type: 'String' },
        locale: { __type: 'String' },
        mediaItems: {
            __type: 'UserToMediaItemConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'UserToMediaItemConnectionWhereArgs',
            },
        },
        name: { __type: 'String' },
        nicename: { __type: 'String' },
        nickname: { __type: 'String' },
        pages: {
            __type: 'UserToPageConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'UserToPageConnectionWhereArgs',
            },
        },
        posts: {
            __type: 'UserToPostConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'UserToPostConnectionWhereArgs',
            },
        },
        registeredDate: { __type: 'String' },
        revisions: {
            __type: 'UserToContentRevisionUnionConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'UserToContentRevisionUnionConnectionWhereArgs',
            },
        },
        roles: {
            __type: 'UserToUserRoleConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        slug: { __type: 'String' },
        uri: { __type: 'String' },
        url: { __type: 'String' },
        userId: { __type: 'Int' },
        username: { __type: 'String' },
    },
    Commenter: {
        __typename: { __type: 'String!' },
        databaseId: { __type: 'Int!' },
        email: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        name: { __type: 'String' },
        url: { __type: 'String' },
    },
    Avatar: {
        __typename: { __type: 'String!' },
        default: { __type: 'String' },
        extraAttr: { __type: 'String' },
        forceDefault: { __type: 'Boolean' },
        foundAvatar: { __type: 'Boolean' },
        height: { __type: 'Int' },
        isRestricted: { __type: 'Boolean' },
        rating: { __type: 'String' },
        scheme: { __type: 'String' },
        size: { __type: 'Int' },
        url: { __type: 'String' },
        width: { __type: 'Int' },
    },
    UserToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    UserToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    Comment: {
        __typename: { __type: 'String!' },
        agent: { __type: 'String' },
        approved: { __type: 'Boolean' },
        author: { __type: 'CommentToCommenterConnectionEdge' },
        authorIp: { __type: 'String' },
        commentId: { __type: 'Int' },
        commentedOn: { __type: 'CommentToContentNodeConnectionEdge' },
        content: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        karma: { __type: 'Int' },
        parent: {
            __type: 'CommentToParentCommentConnectionEdge',
            __args: { where: 'CommentToParentCommentConnectionWhereArgs' },
        },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
        replies: {
            __type: 'CommentToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'CommentToCommentConnectionWhereArgs',
            },
        },
        type: { __type: 'String' },
    },
    CommentToCommenterConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Commenter' },
    },
    CommentToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'ContentNode' },
    },
    CommentToParentCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    CommentToParentCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Comment' },
    },
    CommentToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    CommentToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[CommentToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    CommentToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    UserToEnqueuedScriptConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToEnqueuedScriptConnectionEdge]' },
        nodes: { __type: '[EnqueuedScript]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToEnqueuedScriptConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedScript' },
    },
    UserToEnqueuedStylesheetConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToEnqueuedStylesheetConnectionEdge]' },
        nodes: { __type: '[EnqueuedStylesheet]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToEnqueuedStylesheetConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedStylesheet' },
    },
    UserToMediaItemConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    UserToMediaItemConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToMediaItemConnectionEdge]' },
        nodes: { __type: '[MediaItem]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToMediaItemConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'MediaItem' },
    },
    MediaItem: {
        __typename: { __type: 'String!' },
        altText: { __type: 'String' },
        ancestors: {
            __type: 'HierarchicalContentNodeToContentNodeAncestorsConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeAncestorsConnectionWhereArgs',
            },
        },
        author: { __type: 'NodeWithAuthorToUserConnectionEdge' },
        authorDatabaseId: { __type: 'Int' },
        authorId: { __type: 'ID' },
        caption: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        children: {
            __type: 'HierarchicalContentNodeToContentNodeChildrenConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeChildrenConnectionWhereArgs',
            },
        },
        commentCount: { __type: 'Int' },
        commentStatus: { __type: 'String' },
        comments: {
            __type: 'MediaItemToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'MediaItemToCommentConnectionWhereArgs',
            },
        },
        contentType: { __type: 'ContentNodeToContentTypeConnectionEdge' },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        description: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        desiredSlug: { __type: 'String' },
        editingLockedBy: { __type: 'ContentNodeToEditLockConnectionEdge' },
        enclosure: { __type: 'String' },
        enqueuedScripts: {
            __type: 'ContentNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'ContentNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        fileSize: { __type: 'Int', __args: { size: 'MediaItemSizeEnum' } },
        guid: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isPreview: { __type: 'Boolean' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        lastEditedBy: { __type: 'ContentNodeToEditLastConnectionEdge' },
        link: { __type: 'String' },
        mediaDetails: { __type: 'MediaDetails' },
        mediaItemId: { __type: 'Int!' },
        mediaItemUrl: { __type: 'String' },
        mediaType: { __type: 'String' },
        mimeType: { __type: 'String' },
        modified: { __type: 'String' },
        modifiedGmt: { __type: 'String' },
        parent: {
            __type: 'HierarchicalContentNodeToParentContentNodeConnectionEdge',
        },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
        previewRevisionDatabaseId: { __type: 'Int' },
        previewRevisionId: { __type: 'ID' },
        sizes: { __type: 'String', __args: { size: 'MediaItemSizeEnum' } },
        slug: { __type: 'String' },
        sourceUrl: { __type: 'String', __args: { size: 'MediaItemSizeEnum' } },
        srcSet: { __type: 'String', __args: { size: 'MediaItemSizeEnum' } },
        status: { __type: 'String' },
        template: { __type: 'ContentTemplate' },
        title: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        uri: { __type: 'String' },
    },
    NodeWithTemplate: {
        __typename: { __type: 'String!' },
        template: { __type: 'ContentTemplate' },
    },
    ContentTemplate: {
        __typename: { __type: 'String!' },
        templateName: { __type: 'String' },
    },
    NodeWithTitle: {
        __typename: { __type: 'String!' },
        title: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
    },
    NodeWithAuthor: {
        __typename: { __type: 'String!' },
        author: { __type: 'NodeWithAuthorToUserConnectionEdge' },
        authorDatabaseId: { __type: 'Int' },
        authorId: { __type: 'ID' },
    },
    NodeWithAuthorToUserConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'User' },
    },
    NodeWithComments: {
        __typename: { __type: 'String!' },
        commentCount: { __type: 'Int' },
        commentStatus: { __type: 'String' },
    },
    HierarchicalContentNode: {
        __typename: { __type: 'String!' },
        ancestors: {
            __type: 'HierarchicalContentNodeToContentNodeAncestorsConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeAncestorsConnectionWhereArgs',
            },
        },
        children: {
            __type: 'HierarchicalContentNodeToContentNodeChildrenConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeChildrenConnectionWhereArgs',
            },
        },
        parent: {
            __type: 'HierarchicalContentNodeToParentContentNodeConnectionEdge',
        },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
    },
    HierarchicalContentNodeToContentNodeAncestorsConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    HierarchicalContentNodeToContentNodeAncestorsConnection: {
        __typename: { __type: 'String!' },
        edges: {
            __type: '[HierarchicalContentNodeToContentNodeAncestorsConnectionEdge]',
        },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    HierarchicalContentNodeToContentNodeAncestorsConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    HierarchicalContentNodeToContentNodeChildrenConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    HierarchicalContentNodeToContentNodeChildrenConnection: {
        __typename: { __type: 'String!' },
        edges: {
            __type: '[HierarchicalContentNodeToContentNodeChildrenConnectionEdge]',
        },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    HierarchicalContentNodeToContentNodeChildrenConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    HierarchicalContentNodeToParentContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'ContentNode' },
    },
    MediaItemToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    MediaItemToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[MediaItemToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    MediaItemToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    ContentNodeToEnqueuedScriptConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[ContentNodeToEnqueuedScriptConnectionEdge]' },
        nodes: { __type: '[EnqueuedScript]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    ContentNodeToEnqueuedScriptConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedScript' },
    },
    ContentNodeToEnqueuedStylesheetConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[ContentNodeToEnqueuedStylesheetConnectionEdge]' },
        nodes: { __type: '[EnqueuedStylesheet]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    ContentNodeToEnqueuedStylesheetConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedStylesheet' },
    },
    ContentNodeToEditLastConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'User' },
    },
    MediaDetails: {
        __typename: { __type: 'String!' },
        file: { __type: 'String' },
        height: { __type: 'Int' },
        meta: { __type: 'MediaItemMeta' },
        sizes: { __type: '[MediaSize]' },
        width: { __type: 'Int' },
    },
    MediaItemMeta: {
        __typename: { __type: 'String!' },
        aperture: { __type: 'Float' },
        camera: { __type: 'String' },
        caption: { __type: 'String' },
        copyright: { __type: 'String' },
        createdTimestamp: { __type: 'Int' },
        credit: { __type: 'String' },
        focalLength: { __type: 'Float' },
        iso: { __type: 'Int' },
        keywords: { __type: '[String]' },
        orientation: { __type: 'String' },
        shutterSpeed: { __type: 'Float' },
        title: { __type: 'String' },
    },
    MediaSize: {
        __typename: { __type: 'String!' },
        file: { __type: 'String' },
        fileSize: { __type: 'Int' },
        height: { __type: 'String' },
        mimeType: { __type: 'String' },
        name: { __type: 'String' },
        sourceUrl: { __type: 'String' },
        width: { __type: 'String' },
    },
    UserToPageConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    UserToPageConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToPageConnectionEdge]' },
        nodes: { __type: '[Page]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToPageConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Page' },
    },
    Page: {
        __typename: { __type: 'String!' },
        ancestors: {
            __type: 'HierarchicalContentNodeToContentNodeAncestorsConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeAncestorsConnectionWhereArgs',
            },
        },
        author: { __type: 'NodeWithAuthorToUserConnectionEdge' },
        authorDatabaseId: { __type: 'Int' },
        authorId: { __type: 'ID' },
        children: {
            __type: 'HierarchicalContentNodeToContentNodeChildrenConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'HierarchicalContentNodeToContentNodeChildrenConnectionWhereArgs',
            },
        },
        commentCount: { __type: 'Int' },
        commentStatus: { __type: 'String' },
        comments: {
            __type: 'PageToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PageToCommentConnectionWhereArgs',
            },
        },
        content: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        contentType: { __type: 'ContentNodeToContentTypeConnectionEdge' },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        desiredSlug: { __type: 'String' },
        editingLockedBy: { __type: 'ContentNodeToEditLockConnectionEdge' },
        enclosure: { __type: 'String' },
        enqueuedScripts: {
            __type: 'ContentNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'ContentNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        featuredImage: { __type: 'NodeWithFeaturedImageToMediaItemConnectionEdge' },
        featuredImageDatabaseId: { __type: 'Int' },
        featuredImageId: { __type: 'ID' },
        guid: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isFrontPage: { __type: 'Boolean!' },
        isPostsPage: { __type: 'Boolean!' },
        isPreview: { __type: 'Boolean' },
        isPrivacyPage: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isRevision: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        lastEditedBy: { __type: 'ContentNodeToEditLastConnectionEdge' },
        link: { __type: 'String' },
        menuOrder: { __type: 'Int' },
        modified: { __type: 'String' },
        modifiedGmt: { __type: 'String' },
        pageId: { __type: 'Int!' },
        parent: {
            __type: 'HierarchicalContentNodeToParentContentNodeConnectionEdge',
        },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
        preview: { __type: 'PageToPreviewConnectionEdge' },
        previewRevisionDatabaseId: { __type: 'Int' },
        previewRevisionId: { __type: 'ID' },
        revisionOf: { __type: 'NodeWithRevisionsToContentNodeConnectionEdge' },
        revisions: {
            __type: 'PageToRevisionConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PageToRevisionConnectionWhereArgs',
            },
        },
        slug: { __type: 'String' },
        status: { __type: 'String' },
        template: { __type: 'ContentTemplate' },
        title: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        uri: { __type: 'String' },
    },
    NodeWithContentEditor: {
        __typename: { __type: 'String!' },
        content: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
    },
    NodeWithFeaturedImage: {
        __typename: { __type: 'String!' },
        contentType: { __type: 'ContentNodeToContentTypeConnectionEdge' },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        desiredSlug: { __type: 'String' },
        editingLockedBy: { __type: 'ContentNodeToEditLockConnectionEdge' },
        enclosure: { __type: 'String' },
        enqueuedScripts: {
            __type: 'ContentNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'ContentNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        featuredImage: { __type: 'NodeWithFeaturedImageToMediaItemConnectionEdge' },
        featuredImageDatabaseId: { __type: 'Int' },
        featuredImageId: { __type: 'ID' },
        guid: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isPreview: { __type: 'Boolean' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        lastEditedBy: { __type: 'ContentNodeToEditLastConnectionEdge' },
        link: { __type: 'String' },
        modified: { __type: 'String' },
        modifiedGmt: { __type: 'String' },
        previewRevisionDatabaseId: { __type: 'Int' },
        previewRevisionId: { __type: 'ID' },
        slug: { __type: 'String' },
        status: { __type: 'String' },
        template: { __type: 'ContentTemplate' },
        uri: { __type: 'String' },
    },
    NodeWithFeaturedImageToMediaItemConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'MediaItem' },
    },
    NodeWithRevisions: {
        __typename: { __type: 'String!' },
        isRevision: { __type: 'Boolean' },
        revisionOf: { __type: 'NodeWithRevisionsToContentNodeConnectionEdge' },
    },
    NodeWithRevisionsToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'ContentNode' },
    },
    NodeWithPageAttributes: {
        __typename: { __type: 'String!' },
        menuOrder: { __type: 'Int' },
    },
    PageToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    PageToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PageToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PageToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    PageToPreviewConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Page' },
    },
    PageToRevisionConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    PageToRevisionConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PageToRevisionConnectionEdge]' },
        nodes: { __type: '[Page]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PageToRevisionConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Page' },
    },
    UserToPostConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    UserToPostConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToPostConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToPostConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    Post: {
        __typename: { __type: 'String!' },
        author: { __type: 'NodeWithAuthorToUserConnectionEdge' },
        authorDatabaseId: { __type: 'Int' },
        authorId: { __type: 'ID' },
        categories: {
            __type: 'PostToCategoryConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToCategoryConnectionWhereArgs',
            },
        },
        commentCount: { __type: 'Int' },
        commentStatus: { __type: 'String' },
        comments: {
            __type: 'PostToCommentConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToCommentConnectionWhereArgs',
            },
        },
        content: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        contentType: { __type: 'ContentNodeToContentTypeConnectionEdge' },
        databaseId: { __type: 'Int!' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        desiredSlug: { __type: 'String' },
        editingLockedBy: { __type: 'ContentNodeToEditLockConnectionEdge' },
        enclosure: { __type: 'String' },
        enqueuedScripts: {
            __type: 'ContentNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'ContentNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        excerpt: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        featuredImage: { __type: 'NodeWithFeaturedImageToMediaItemConnectionEdge' },
        featuredImageDatabaseId: { __type: 'Int' },
        featuredImageId: { __type: 'ID' },
        guid: { __type: 'String' },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isPreview: { __type: 'Boolean' },
        isRestricted: { __type: 'Boolean' },
        isRevision: { __type: 'Boolean' },
        isSticky: { __type: 'Boolean!' },
        isTermNode: { __type: 'Boolean!' },
        lastEditedBy: { __type: 'ContentNodeToEditLastConnectionEdge' },
        link: { __type: 'String' },
        modified: { __type: 'String' },
        modifiedGmt: { __type: 'String' },
        pingStatus: { __type: 'String' },
        pinged: { __type: '[String]' },
        postFormats: {
            __type: 'PostToPostFormatConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToPostFormatConnectionWhereArgs',
            },
        },
        postId: { __type: 'Int!' },
        preview: { __type: 'PostToPreviewConnectionEdge' },
        previewRevisionDatabaseId: { __type: 'Int' },
        previewRevisionId: { __type: 'ID' },
        revisionOf: { __type: 'NodeWithRevisionsToContentNodeConnectionEdge' },
        revisions: {
            __type: 'PostToRevisionConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToRevisionConnectionWhereArgs',
            },
        },
        slug: { __type: 'String' },
        status: { __type: 'String' },
        tags: {
            __type: 'PostToTagConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToTagConnectionWhereArgs',
            },
        },
        template: { __type: 'ContentTemplate' },
        terms: {
            __type: 'PostToTermNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostToTermNodeConnectionWhereArgs',
            },
        },
        title: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
        toPing: { __type: '[String]' },
        uri: { __type: 'String' },
    },
    NodeWithExcerpt: {
        __typename: { __type: 'String!' },
        excerpt: {
            __type: 'String',
            __args: { format: 'PostObjectFieldFormatEnum' },
        },
    },
    NodeWithTrackbacks: {
        __typename: { __type: 'String!' },
        pingStatus: { __type: 'String' },
        pinged: { __type: '[String]' },
        toPing: { __type: '[String]' },
    },
    PostToCategoryConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    PostToCategoryConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToCategoryConnectionEdge]' },
        nodes: { __type: '[Category]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToCategoryConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Category' },
    },
    PostToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    PostToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    PostToPostFormatConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    PostToPostFormatConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToPostFormatConnectionEdge]' },
        nodes: { __type: '[PostFormat]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToPostFormatConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'PostFormat' },
    },
    PostFormat: {
        __typename: { __type: 'String!' },
        contentNodes: {
            __type: 'PostFormatToContentNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostFormatToContentNodeConnectionWhereArgs',
            },
        },
        count: { __type: 'Int' },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        enqueuedScripts: {
            __type: 'TermNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'TermNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        link: { __type: 'String' },
        name: { __type: 'String' },
        postFormatId: { __type: 'Int' },
        posts: {
            __type: 'PostFormatToPostConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'PostFormatToPostConnectionWhereArgs',
            },
        },
        slug: { __type: 'String' },
        taxonomy: { __type: 'PostFormatToTaxonomyConnectionEdge' },
        termGroupId: { __type: 'Int' },
        termTaxonomyId: { __type: 'Int' },
        uri: { __type: 'String' },
    },
    PostFormatToContentNodeConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypesOfPostFormatEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    PostFormatToContentNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostFormatToContentNodeConnectionEdge]' },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostFormatToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    PostFormatToPostConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    PostFormatToPostConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostFormatToPostConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostFormatToPostConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    PostFormatToTaxonomyConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Taxonomy' },
    },
    PostToPreviewConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Post' },
    },
    PostToRevisionConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    PostToRevisionConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToRevisionConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToRevisionConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    PostToTagConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    PostToTagConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToTagConnectionEdge]' },
        nodes: { __type: '[Tag]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToTagConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Tag' },
    },
    Tag: {
        __typename: { __type: 'String!' },
        contentNodes: {
            __type: 'TagToContentNodeConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'TagToContentNodeConnectionWhereArgs',
            },
        },
        count: { __type: 'Int' },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        enqueuedScripts: {
            __type: 'TermNodeToEnqueuedScriptConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        enqueuedStylesheets: {
            __type: 'TermNodeToEnqueuedStylesheetConnection',
            __args: { first: 'Int', last: 'Int', after: 'String', before: 'String' },
        },
        id: { __type: 'ID!' },
        isContentNode: { __type: 'Boolean!' },
        isRestricted: { __type: 'Boolean' },
        isTermNode: { __type: 'Boolean!' },
        link: { __type: 'String' },
        name: { __type: 'String' },
        posts: {
            __type: 'TagToPostConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'TagToPostConnectionWhereArgs',
            },
        },
        slug: { __type: 'String' },
        tagId: { __type: 'Int' },
        taxonomy: { __type: 'TagToTaxonomyConnectionEdge' },
        termGroupId: { __type: 'Int' },
        termTaxonomyId: { __type: 'Int' },
        uri: { __type: 'String' },
    },
    TagToContentNodeConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypesOfTagEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    TagToContentNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[TagToContentNodeConnectionEdge]' },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    TagToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    TagToPostConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    TagToPostConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[TagToPostConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    TagToPostConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    TagToTaxonomyConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Taxonomy' },
    },
    PostToTermNodeConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        taxonomies: { __type: '[TaxonomyEnum]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    PostToTermNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[PostToTermNodeConnectionEdge]' },
        nodes: { __type: '[TermNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    PostToTermNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'TermNode' },
    },
    UserToContentRevisionUnionConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    UserToContentRevisionUnionConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToContentRevisionUnionConnectionEdge]' },
        nodes: { __type: '[ContentRevisionUnion]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToContentRevisionUnionConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentRevisionUnion' },
    },
    UserToUserRoleConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[UserToUserRoleConnectionEdge]' },
        nodes: { __type: '[UserRole]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    UserToUserRoleConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'UserRole' },
    },
    UserRole: {
        __typename: { __type: 'String!' },
        capabilities: { __type: '[String]' },
        displayName: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        name: { __type: 'String' },
    },
    CategoryToParentCategoryConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Category' },
    },
    CategoryToPostConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    CategoryToPostConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[CategoryToPostConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    CategoryToPostConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    CategoryToTaxonomyConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Taxonomy' },
    },
    RootQueryToCommentConnectionWhereArgs: {
        authorEmail: { __type: 'String' },
        authorIn: { __type: '[ID]' },
        authorNotIn: { __type: '[ID]' },
        authorUrl: { __type: 'String' },
        commentIn: { __type: '[ID]' },
        commentNotIn: { __type: '[ID]' },
        commentType: { __type: 'String' },
        commentTypeIn: { __type: '[String]' },
        commentTypeNotIn: { __type: 'String' },
        contentAuthor: { __type: '[ID]' },
        contentAuthorIn: { __type: '[ID]' },
        contentAuthorNotIn: { __type: '[ID]' },
        contentId: { __type: 'ID' },
        contentIdIn: { __type: '[ID]' },
        contentIdNotIn: { __type: '[ID]' },
        contentName: { __type: 'String' },
        contentParent: { __type: 'Int' },
        contentStatus: { __type: '[PostStatusEnum]' },
        contentType: { __type: '[ContentTypeEnum]' },
        includeUnapproved: { __type: '[ID]' },
        karma: { __type: 'Int' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'CommentsConnectionOrderbyEnum' },
        parent: { __type: 'Int' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        search: { __type: 'String' },
        status: { __type: 'String' },
        userId: { __type: 'ID' },
    },
    RootQueryToCommentConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToCommentConnectionEdge]' },
        nodes: { __type: '[Comment]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToCommentConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Comment' },
    },
    RootQueryToContentNodeConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    RootQueryToContentNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToContentNodeConnectionEdge]' },
        nodes: { __type: '[ContentNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToContentNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentNode' },
    },
    RootQueryToContentTypeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToContentTypeConnectionEdge]' },
        nodes: { __type: '[ContentType]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToContentTypeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentType' },
    },
    DiscussionSettings: {
        __typename: { __type: 'String!' },
        defaultCommentStatus: { __type: 'String' },
        defaultPingStatus: { __type: 'String' },
    },
    GeneralSettings: {
        __typename: { __type: 'String!' },
        dateFormat: { __type: 'String' },
        description: { __type: 'String' },
        email: { __type: 'String' },
        language: { __type: 'String' },
        startOfWeek: { __type: 'Int' },
        timeFormat: { __type: 'String' },
        timezone: { __type: 'String' },
        title: { __type: 'String' },
        url: { __type: 'String' },
    },
    RootQueryToMediaItemConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    RootQueryToMediaItemConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToMediaItemConnectionEdge]' },
        nodes: { __type: '[MediaItem]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToMediaItemConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'MediaItem' },
    },
    Menu: {
        __typename: { __type: 'String!' },
        count: { __type: 'Int' },
        databaseId: { __type: 'Int!' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        locations: { __type: '[MenuLocationEnum]' },
        menuId: { __type: 'Int' },
        menuItems: {
            __type: 'MenuToMenuItemConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'MenuToMenuItemConnectionWhereArgs',
            },
        },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    MenuToMenuItemConnectionWhereArgs: {
        id: { __type: 'Int' },
        location: { __type: 'MenuLocationEnum' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
    },
    MenuToMenuItemConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[MenuToMenuItemConnectionEdge]' },
        nodes: { __type: '[MenuItem]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    MenuToMenuItemConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'MenuItem' },
    },
    MenuItem: {
        __typename: { __type: 'String!' },
        childItems: {
            __type: 'MenuItemToMenuItemConnection',
            __args: {
                first: 'Int',
                last: 'Int',
                after: 'String',
                before: 'String',
                where: 'MenuItemToMenuItemConnectionWhereArgs',
            },
        },
        connectedNode: { __type: 'MenuItemToMenuItemLinkableConnectionEdge' },
        connectedObject: { __type: 'MenuItemObjectUnion' },
        cssClasses: { __type: '[String]' },
        databaseId: { __type: 'Int!' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        label: { __type: 'String' },
        linkRelationship: { __type: 'String' },
        locations: { __type: '[MenuLocationEnum]' },
        menu: { __type: 'MenuItemToMenuConnectionEdge' },
        menuItemId: { __type: 'Int' },
        order: { __type: 'Int' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
        path: { __type: 'String' },
        target: { __type: 'String' },
        title: { __type: 'String' },
        url: { __type: 'String' },
    },
    MenuItemToMenuItemConnectionWhereArgs: {
        id: { __type: 'Int' },
        location: { __type: 'MenuLocationEnum' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
    },
    MenuItemToMenuItemConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[MenuItemToMenuItemConnectionEdge]' },
        nodes: { __type: '[MenuItem]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    MenuItemToMenuItemConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'MenuItem' },
    },
    MenuItemToMenuItemLinkableConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'MenuItemLinkable' },
    },
    MenuItemToMenuConnectionEdge: {
        __typename: { __type: 'String!' },
        node: { __type: 'Menu' },
    },
    RootQueryToMenuItemConnectionWhereArgs: {
        id: { __type: 'Int' },
        location: { __type: 'MenuLocationEnum' },
        parentDatabaseId: { __type: 'Int' },
        parentId: { __type: 'ID' },
    },
    RootQueryToMenuItemConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToMenuItemConnectionEdge]' },
        nodes: { __type: '[MenuItem]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToMenuItemConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'MenuItem' },
    },
    RootQueryToMenuConnectionWhereArgs: {
        id: { __type: 'Int' },
        location: { __type: 'MenuLocationEnum' },
        slug: { __type: 'String' },
    },
    RootQueryToMenuConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToMenuConnectionEdge]' },
        nodes: { __type: '[Menu]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToMenuConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Menu' },
    },
    RootQueryToPageConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    RootQueryToPageConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToPageConnectionEdge]' },
        nodes: { __type: '[Page]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToPageConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Page' },
    },
    Plugin: {
        __typename: { __type: 'String!' },
        author: { __type: 'String' },
        authorUri: { __type: 'String' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        name: { __type: 'String' },
        path: { __type: 'String' },
        pluginUri: { __type: 'String' },
        version: { __type: 'String' },
    },
    RootQueryToPluginConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToPluginConnectionEdge]' },
        nodes: { __type: '[Plugin]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToPluginConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Plugin' },
    },
    RootQueryToPostFormatConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    RootQueryToPostFormatConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToPostFormatConnectionEdge]' },
        nodes: { __type: '[PostFormat]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToPostFormatConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'PostFormat' },
    },
    RootQueryToPostConnectionWhereArgs: {
        author: { __type: 'Int' },
        authorIn: { __type: '[ID]' },
        authorName: { __type: 'String' },
        authorNotIn: { __type: '[ID]' },
        categoryId: { __type: 'Int' },
        categoryIn: { __type: '[ID]' },
        categoryName: { __type: 'String' },
        categoryNotIn: { __type: '[ID]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        tag: { __type: 'String' },
        tagId: { __type: 'String' },
        tagIn: { __type: '[ID]' },
        tagNotIn: { __type: '[ID]' },
        tagSlugAnd: { __type: '[String]' },
        tagSlugIn: { __type: '[String]' },
        title: { __type: 'String' },
    },
    RootQueryToPostConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToPostConnectionEdge]' },
        nodes: { __type: '[Post]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToPostConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Post' },
    },
    ReadingSettings: {
        __typename: { __type: 'String!' },
        postsPerPage: { __type: 'Int' },
    },
    RootQueryToEnqueuedScriptConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToEnqueuedScriptConnectionEdge]' },
        nodes: { __type: '[EnqueuedScript]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToEnqueuedScriptConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedScript' },
    },
    RootQueryToEnqueuedStylesheetConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToEnqueuedStylesheetConnectionEdge]' },
        nodes: { __type: '[EnqueuedStylesheet]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToEnqueuedStylesheetConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'EnqueuedStylesheet' },
    },
    RootQueryToContentRevisionUnionConnectionWhereArgs: {
        contentTypes: { __type: '[ContentTypeEnum]' },
        dateQuery: { __type: 'DateQueryInput' },
        hasPassword: { __type: 'Boolean' },
        id: { __type: 'Int' },
        in: { __type: '[ID]' },
        mimeType: { __type: 'MimeTypeEnum' },
        name: { __type: 'String' },
        nameIn: { __type: '[String]' },
        notIn: { __type: '[ID]' },
        orderby: { __type: '[PostObjectsConnectionOrderbyInput]' },
        parent: { __type: 'ID' },
        parentIn: { __type: '[ID]' },
        parentNotIn: { __type: '[ID]' },
        password: { __type: 'String' },
        search: { __type: 'String' },
        stati: { __type: '[PostStatusEnum]' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    RootQueryToContentRevisionUnionConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToContentRevisionUnionConnectionEdge]' },
        nodes: { __type: '[ContentRevisionUnion]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToContentRevisionUnionConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'ContentRevisionUnion' },
    },
    RootQueryToTagConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    RootQueryToTagConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToTagConnectionEdge]' },
        nodes: { __type: '[Tag]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToTagConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Tag' },
    },
    RootQueryToTaxonomyConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToTaxonomyConnectionEdge]' },
        nodes: { __type: '[Taxonomy]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToTaxonomyConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Taxonomy' },
    },
    RootQueryToTermNodeConnectionWhereArgs: {
        cacheDomain: { __type: 'String' },
        childOf: { __type: 'Int' },
        childless: { __type: 'Boolean' },
        descriptionLike: { __type: 'String' },
        exclude: { __type: '[ID]' },
        excludeTree: { __type: '[ID]' },
        hideEmpty: { __type: 'Boolean' },
        hierarchical: { __type: 'Boolean' },
        include: { __type: '[ID]' },
        name: { __type: '[String]' },
        nameLike: { __type: 'String' },
        objectIds: { __type: '[ID]' },
        order: { __type: 'OrderEnum' },
        orderby: { __type: 'TermObjectsConnectionOrderbyEnum' },
        padCounts: { __type: 'Boolean' },
        parent: { __type: 'Int' },
        search: { __type: 'String' },
        slug: { __type: '[String]' },
        taxonomies: { __type: '[TaxonomyEnum]' },
        termTaxonomId: { __type: '[ID]' },
        updateTermMetaCache: { __type: 'Boolean' },
    },
    RootQueryToTermNodeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToTermNodeConnectionEdge]' },
        nodes: { __type: '[TermNode]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToTermNodeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'TermNode' },
    },
    Theme: {
        __typename: { __type: 'String!' },
        author: { __type: 'String' },
        authorUri: { __type: 'String' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        name: { __type: 'String' },
        screenshot: { __type: 'String' },
        slug: { __type: 'String' },
        tags: { __type: '[String]' },
        themeUri: { __type: 'String' },
        version: { __type: 'String' },
    },
    RootQueryToThemeConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToThemeConnectionEdge]' },
        nodes: { __type: '[Theme]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToThemeConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'Theme' },
    },
    RootQueryToUserRoleConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToUserRoleConnectionEdge]' },
        nodes: { __type: '[UserRole]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToUserRoleConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'UserRole' },
    },
    RootQueryToUserConnectionWhereArgs: {
        exclude: { __type: '[Int]' },
        hasPublishedPosts: { __type: '[ContentTypeEnum]' },
        include: { __type: '[Int]' },
        login: { __type: 'String' },
        loginIn: { __type: '[String]' },
        loginNotIn: { __type: '[String]' },
        nicename: { __type: 'String' },
        nicenameIn: { __type: '[String]' },
        nicenameNotIn: { __type: '[String]' },
        orderby: { __type: '[UsersConnectionOrderbyInput]' },
        role: { __type: 'UserRoleEnum' },
        roleIn: { __type: '[UserRoleEnum]' },
        roleNotIn: { __type: '[UserRoleEnum]' },
        search: { __type: 'String' },
        searchColumns: { __type: '[UsersConnectionSearchColumnEnum]' },
    },
    UsersConnectionOrderbyInput: {
        field: { __type: 'UsersConnectionOrderbyEnum!' },
        order: { __type: 'OrderEnum' },
    },
    RootQueryToUserConnection: {
        __typename: { __type: 'String!' },
        edges: { __type: '[RootQueryToUserConnectionEdge]' },
        nodes: { __type: '[User]' },
        pageInfo: { __type: 'WPPageInfo' },
    },
    RootQueryToUserConnectionEdge: {
        __typename: { __type: 'String!' },
        cursor: { __type: 'String' },
        node: { __type: 'User' },
    },
    WritingSettings: {
        __typename: { __type: 'String!' },
        defaultCategory: { __type: 'Int' },
        defaultPostFormat: { __type: 'String' },
        useSmilies: { __type: 'Boolean' },
    },
    CreateCategoryInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        name: { __type: 'String!' },
        parentId: { __type: 'ID' },
        slug: { __type: 'String' },
    },
    CreateCategoryPayload: {
        __typename: { __type: 'String!' },
        category: { __type: 'Category' },
        clientMutationId: { __type: 'String' },
    },
    CreateCommentInput: {
        approved: { __type: 'String' },
        author: { __type: 'String' },
        authorEmail: { __type: 'String' },
        authorUrl: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        commentOn: { __type: 'Int' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        parent: { __type: 'ID' },
        type: { __type: 'String' },
    },
    CreateCommentPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        comment: { __type: 'Comment' },
        success: { __type: 'Boolean' },
    },
    CreateMediaItemInput: {
        altText: { __type: 'String' },
        authorId: { __type: 'ID' },
        caption: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        description: { __type: 'String' },
        filePath: { __type: 'String' },
        fileType: { __type: 'MimeTypeEnum' },
        parentId: { __type: 'ID' },
        pingStatus: { __type: 'String' },
        slug: { __type: 'String' },
        status: { __type: 'MediaItemStatusEnum' },
        title: { __type: 'String' },
    },
    CreateMediaItemPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        mediaItem: { __type: 'MediaItem' },
    },
    CreatePageInput: {
        authorId: { __type: 'ID' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        menuOrder: { __type: 'Int' },
        parentId: { __type: 'ID' },
        password: { __type: 'String' },
        slug: { __type: 'String' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    CreatePagePayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        page: { __type: 'Page' },
    },
    CreatePostInput: {
        authorId: { __type: 'ID' },
        categories: { __type: 'PostCategoriesInput' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        excerpt: { __type: 'String' },
        menuOrder: { __type: 'Int' },
        password: { __type: 'String' },
        pingStatus: { __type: 'String' },
        pinged: { __type: '[String]' },
        postFormats: { __type: 'PostPostFormatsInput' },
        slug: { __type: 'String' },
        status: { __type: 'PostStatusEnum' },
        tags: { __type: 'PostTagsInput' },
        title: { __type: 'String' },
        toPing: { __type: '[String]' },
    },
    PostCategoriesInput: {
        append: { __type: 'Boolean' },
        nodes: { __type: '[PostCategoriesNodeInput]' },
    },
    PostCategoriesNodeInput: {
        description: { __type: 'String' },
        id: { __type: 'ID' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    PostPostFormatsInput: {
        append: { __type: 'Boolean' },
        nodes: { __type: '[PostPostFormatsNodeInput]' },
    },
    PostPostFormatsNodeInput: {
        description: { __type: 'String' },
        id: { __type: 'ID' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    PostTagsInput: {
        append: { __type: 'Boolean' },
        nodes: { __type: '[PostTagsNodeInput]' },
    },
    PostTagsNodeInput: {
        description: { __type: 'String' },
        id: { __type: 'ID' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    CreatePostPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        post: { __type: 'Post' },
    },
    CreatePostFormatInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        name: { __type: 'String!' },
        slug: { __type: 'String' },
    },
    CreatePostFormatPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        postFormat: { __type: 'PostFormat' },
    },
    CreateTagInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        name: { __type: 'String!' },
        slug: { __type: 'String' },
    },
    CreateTagPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        tag: { __type: 'Tag' },
    },
    CreateUserInput: {
        aim: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        displayName: { __type: 'String' },
        email: { __type: 'String' },
        firstName: { __type: 'String' },
        jabber: { __type: 'String' },
        lastName: { __type: 'String' },
        locale: { __type: 'String' },
        nicename: { __type: 'String' },
        nickname: { __type: 'String' },
        password: { __type: 'String' },
        registered: { __type: 'String' },
        richEditing: { __type: 'String' },
        roles: { __type: '[String]' },
        username: { __type: 'String!' },
        websiteUrl: { __type: 'String' },
        yim: { __type: 'String' },
    },
    CreateUserPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        user: { __type: 'User' },
    },
    DeleteCategoryInput: {
        clientMutationId: { __type: 'String' },
        id: { __type: 'ID!' },
    },
    DeleteCategoryPayload: {
        __typename: { __type: 'String!' },
        category: { __type: 'Category' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
    },
    DeleteCommentInput: {
        clientMutationId: { __type: 'String' },
        forceDelete: { __type: 'Boolean' },
        id: { __type: 'ID!' },
    },
    DeleteCommentPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        comment: { __type: 'Comment' },
        deletedId: { __type: 'ID' },
    },
    DeleteMediaItemInput: {
        clientMutationId: { __type: 'String' },
        forceDelete: { __type: 'Boolean' },
        id: { __type: 'ID!' },
    },
    DeleteMediaItemPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        mediaItem: { __type: 'MediaItem' },
    },
    DeletePageInput: {
        clientMutationId: { __type: 'String' },
        forceDelete: { __type: 'Boolean' },
        id: { __type: 'ID!' },
    },
    DeletePagePayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        page: { __type: 'Page' },
    },
    DeletePostInput: {
        clientMutationId: { __type: 'String' },
        forceDelete: { __type: 'Boolean' },
        id: { __type: 'ID!' },
    },
    DeletePostPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        post: { __type: 'Post' },
    },
    DeletePostFormatInput: {
        clientMutationId: { __type: 'String' },
        id: { __type: 'ID!' },
    },
    DeletePostFormatPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        postFormat: { __type: 'PostFormat' },
    },
    DeleteTagInput: {
        clientMutationId: { __type: 'String' },
        id: { __type: 'ID!' },
    },
    DeleteTagPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        tag: { __type: 'Tag' },
    },
    DeleteUserInput: {
        clientMutationId: { __type: 'String' },
        id: { __type: 'ID!' },
        reassignId: { __type: 'ID' },
    },
    DeleteUserPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        deletedId: { __type: 'ID' },
        user: { __type: 'User' },
    },
    RegisterUserInput: {
        aim: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        displayName: { __type: 'String' },
        email: { __type: 'String' },
        firstName: { __type: 'String' },
        jabber: { __type: 'String' },
        lastName: { __type: 'String' },
        locale: { __type: 'String' },
        nicename: { __type: 'String' },
        nickname: { __type: 'String' },
        password: { __type: 'String' },
        registered: { __type: 'String' },
        richEditing: { __type: 'String' },
        username: { __type: 'String!' },
        websiteUrl: { __type: 'String' },
        yim: { __type: 'String' },
    },
    RegisterUserPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        user: { __type: 'User' },
    },
    ResetUserPasswordInput: {
        clientMutationId: { __type: 'String' },
        key: { __type: 'String' },
        login: { __type: 'String' },
        password: { __type: 'String' },
    },
    ResetUserPasswordPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        user: { __type: 'User' },
    },
    RestoreCommentInput: {
        clientMutationId: { __type: 'String' },
        id: { __type: 'ID!' },
    },
    RestoreCommentPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        comment: { __type: 'Comment' },
        restoredId: { __type: 'ID' },
    },
    SendPasswordResetEmailInput: {
        clientMutationId: { __type: 'String' },
        username: { __type: 'String!' },
    },
    SendPasswordResetEmailPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        user: { __type: 'User' },
    },
    UpdateCategoryInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        name: { __type: 'String' },
        parentId: { __type: 'ID' },
        slug: { __type: 'String' },
    },
    UpdateCategoryPayload: {
        __typename: { __type: 'String!' },
        category: { __type: 'Category' },
        clientMutationId: { __type: 'String' },
    },
    UpdateCommentInput: {
        approved: { __type: 'String' },
        author: { __type: 'String' },
        authorEmail: { __type: 'String' },
        authorUrl: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        commentOn: { __type: 'Int' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        id: { __type: 'ID!' },
        parent: { __type: 'ID' },
        type: { __type: 'String' },
    },
    UpdateCommentPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        comment: { __type: 'Comment' },
        success: { __type: 'Boolean' },
    },
    UpdateMediaItemInput: {
        altText: { __type: 'String' },
        authorId: { __type: 'ID' },
        caption: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        date: { __type: 'String' },
        dateGmt: { __type: 'String' },
        description: { __type: 'String' },
        filePath: { __type: 'String' },
        fileType: { __type: 'MimeTypeEnum' },
        id: { __type: 'ID!' },
        parentId: { __type: 'ID' },
        pingStatus: { __type: 'String' },
        slug: { __type: 'String' },
        status: { __type: 'MediaItemStatusEnum' },
        title: { __type: 'String' },
    },
    UpdateMediaItemPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        mediaItem: { __type: 'MediaItem' },
    },
    UpdatePageInput: {
        authorId: { __type: 'ID' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        id: { __type: 'ID!' },
        menuOrder: { __type: 'Int' },
        parentId: { __type: 'ID' },
        password: { __type: 'String' },
        slug: { __type: 'String' },
        status: { __type: 'PostStatusEnum' },
        title: { __type: 'String' },
    },
    UpdatePagePayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        page: { __type: 'Page' },
    },
    UpdatePostInput: {
        authorId: { __type: 'ID' },
        categories: { __type: 'PostCategoriesInput' },
        clientMutationId: { __type: 'String' },
        commentStatus: { __type: 'String' },
        content: { __type: 'String' },
        date: { __type: 'String' },
        excerpt: { __type: 'String' },
        id: { __type: 'ID!' },
        menuOrder: { __type: 'Int' },
        password: { __type: 'String' },
        pingStatus: { __type: 'String' },
        pinged: { __type: '[String]' },
        postFormats: { __type: 'PostPostFormatsInput' },
        slug: { __type: 'String' },
        status: { __type: 'PostStatusEnum' },
        tags: { __type: 'PostTagsInput' },
        title: { __type: 'String' },
        toPing: { __type: '[String]' },
    },
    UpdatePostPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        post: { __type: 'Post' },
    },
    UpdatePostFormatInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    UpdatePostFormatPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        postFormat: { __type: 'PostFormat' },
    },
    UpdateSettingsInput: {
        clientMutationId: { __type: 'String' },
        discussionSettingsDefaultCommentStatus: { __type: 'String' },
        discussionSettingsDefaultPingStatus: { __type: 'String' },
        generalSettingsDateFormat: { __type: 'String' },
        generalSettingsDescription: { __type: 'String' },
        generalSettingsEmail: { __type: 'String' },
        generalSettingsLanguage: { __type: 'String' },
        generalSettingsStartOfWeek: { __type: 'Int' },
        generalSettingsTimeFormat: { __type: 'String' },
        generalSettingsTimezone: { __type: 'String' },
        generalSettingsTitle: { __type: 'String' },
        generalSettingsUrl: { __type: 'String' },
        readingSettingsPostsPerPage: { __type: 'Int' },
        writingSettingsDefaultCategory: { __type: 'Int' },
        writingSettingsDefaultPostFormat: { __type: 'String' },
        writingSettingsUseSmilies: { __type: 'Boolean' },
    },
    UpdateSettingsPayload: {
        __typename: { __type: 'String!' },
        allSettings: { __type: 'Settings' },
        clientMutationId: { __type: 'String' },
        discussionSettings: { __type: 'DiscussionSettings' },
        generalSettings: { __type: 'GeneralSettings' },
        readingSettings: { __type: 'ReadingSettings' },
        writingSettings: { __type: 'WritingSettings' },
    },
    UpdateTagInput: {
        aliasOf: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        id: { __type: 'ID!' },
        name: { __type: 'String' },
        slug: { __type: 'String' },
    },
    UpdateTagPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        tag: { __type: 'Tag' },
    },
    UpdateUserInput: {
        aim: { __type: 'String' },
        clientMutationId: { __type: 'String' },
        description: { __type: 'String' },
        displayName: { __type: 'String' },
        email: { __type: 'String' },
        firstName: { __type: 'String' },
        id: { __type: 'ID!' },
        jabber: { __type: 'String' },
        lastName: { __type: 'String' },
        locale: { __type: 'String' },
        nicename: { __type: 'String' },
        nickname: { __type: 'String' },
        password: { __type: 'String' },
        registered: { __type: 'String' },
        richEditing: { __type: 'String' },
        roles: { __type: '[String]' },
        websiteUrl: { __type: 'String' },
        yim: { __type: 'String' },
    },
    UpdateUserPayload: {
        __typename: { __type: 'String!' },
        clientMutationId: { __type: 'String' },
        user: { __type: 'User' },
    },
    CommentAuthor: {
        __typename: { __type: 'String!' },
        databaseId: { __type: 'Int!' },
        email: { __type: 'String' },
        id: { __type: 'ID!' },
        isRestricted: { __type: 'Boolean' },
        name: { __type: 'String' },
        url: { __type: 'String' },
    },
    DefaultTemplate: {
        __typename: { __type: 'String!' },
        templateName: { __type: 'String' },
    },
    [gqty_1.SchemaUnionsKey]: {
        ContentRevisionUnion: ['Post', 'Page'],
        MenuItemObjectUnion: ['Post', 'Page', 'Category', 'Tag', 'PostFormat'],
    },
};
