// gqty/Selection/selection.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

exports.SelectionType = void 0;
(function(SelectionType2) {
  SelectionType2[SelectionType2["Query"] = 0] = "Query";
  SelectionType2[SelectionType2["Mutation"] = 1] = "Mutation";
  SelectionType2[SelectionType2["Subscription"] = 2] = "Subscription";
})(exports.SelectionType || (exports.SelectionType = {}));
class Selection {
  constructor({
    key,
    prevSelection,
    args,
    argTypes,
    type,
    alias,
    unions,
    id
  }) {
    this.cachePath = [];
    this.id = id + "";
    this.key = key;
    const pathKey = alias || key;
    this.cachePath = prevSelection ? [...prevSelection.cachePath, pathKey] : [pathKey];
    this.pathString = prevSelection ? prevSelection.pathString + "." + pathKey : pathKey.toString();
    const prevSelectionsList = (prevSelection == null ? void 0 : prevSelection.selectionsList) || [];
    this.selectionsList = [...prevSelectionsList, this];
    const prevNoSelectionsList = (prevSelection == null ? void 0 : prevSelection.noIndexSelections) || [];
    this.noIndexSelections = typeof key === "string" ? [...prevNoSelectionsList, this] : prevNoSelectionsList;
    if (this.selectionsList.length === this.noIndexSelections.length) {
      this.noIndexSelections = this.selectionsList;
    }
    this.alias = alias;
    this.args = args;
    this.argTypes = argTypes;
    this.unions = unions;
    this.type = type || (prevSelection == null ? void 0 : prevSelection.type) || 0;
  }
}

exports.Selection = Selection;
