// @gqty/react/query/hoc.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const React = require('react');
const common = require('../common.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

const React__default = /*#__PURE__*/_interopDefaultLegacy(React);

function createGraphqlHOC({ scheduler, eventHandler, interceptorManager }, {
  defaults: {
    suspense: defaultSuspense,
    staleWhileRevalidate: defaultStaleWhileRevalidate
  }
}) {
  const graphql = function graphql2(component, {
    suspense = defaultSuspense,
    staleWhileRevalidate = defaultStaleWhileRevalidate,
    onError
  } = {}) {
    const withGraphQL = function WithGraphQL(props) {
      var _a;
      const { fetchingPromise, unsubscribe } = common.useInterceptSelections({
        interceptorManager,
        eventHandler,
        scheduler,
        staleWhileRevalidate,
        onError
      });
      let returnValue = null;
      try {
        returnValue = (_a = component(props)) != null ? _a : null;
      } finally {
        unsubscribe();
      }
      if (suspense && fetchingPromise.current) {
        let Suspend = function() {
          if (!fetchingPromise.current)
            return null;
          throw fetchingPromise.current;
        };
        const value = /* @__PURE__ */ React__default['default'].createElement(React__default['default'].Fragment, null, returnValue, /* @__PURE__ */ React__default['default'].createElement(Suspend, null));
        if (typeof suspense === "object") {
          return /* @__PURE__ */ React__default['default'].createElement(React.Suspense, {
            fallback: suspense.fallback,
            children: value
          });
        }
        return value;
      }
      return returnValue;
    };
    withGraphQL.displayName = `GraphQLComponent(${(component == null ? void 0 : component.displayName) || (component == null ? void 0 : component.name) || "Anonymous"})${Date.now}`;
    return withGraphQL;
  };
  return graphql;
}

exports.createGraphqlHOC = createGraphqlHOC;
